NumLib/Fem/ShapeFunction/ShapeQuad9-impl.h

```diff
@@ -9,44 +9,42 @@
 
 namespace NumLib
 {
-
 template <class T_X, class T_N>
-void ShapeQuad9::computeShapeFunction(const T_X &r, T_N &N)
+void ShapeQuad9::computeShapeFunction(const T_X& r, T_N& N)
 {
-    N[8] = (1.0 - r[0] * r[0]) * ( 1.0 - r[1] * r[1]);
-    N[7] = 0.5 * (1.0 - r[1] * r[1]) * (1.0 + r[0]) - 0.5 * N[8];
-    N[6] = 0.5 * (1.0 - r[0] * r[0]) * (1.0 - r[1]) - 0.5 * N[8];
-    N[5] = 0.5 * (1.0 - r[1] * r[1]) * (1.0 - r[0]) - 0.5 * N[8];
-    N[4] = 0.5 * (1.0 - r[0] * r[0]) * (1.0 + r[1]) - 0.5 * N[8];
-    N[3] = 0.25 * (1.0 + r[0]) * (1.0 - r[1]) - 0.5 * N[6] - 0.5 * N[7] - 0.25 * N[8];
-    N[2] = 0.25 * (1.0 - r[0]) * (1.0 - r[1]) - 0.5 * N[5] - 0.5 * N[6] - 0.25 * N[8];
-    N[1] = 0.25 * (1.0 - r[0]) * (1.0 + r[1]) - 0.5 * N[4] - 0.5 * N[5] - 0.25 * N[8];
-    N[0] = 0.25 * (1.0 + r[0]) * (1.0 + r[1]) - 0.5 * N[4] - 0.5 * N[7] - 0.25 * N[8];
+    N[0] = r[0] * (r[0] - 1) * r[1] * (r[1] - 1) / 4;
+    N[1] = r[0] * (r[0] + 1) * r[1] * (r[1] - 1) / 4;
+    N[2] = r[0] * (r[0] + 1) * r[1] * (r[1] + 1) / 4;
+    N[3] = r[0] * (r[0] - 1) * r[1] * (r[1] + 1) / 4;
+    N[4] = r[1] * (r[1] - 1) * (1 - r[0] * r[0]) / 2;
+    N[5] = r[0] * (r[0] + 1) * (1 - r[1] * r[1]) / 2;
+    N[6] = r[1] * (r[1] + 1) * (1 - r[0] * r[0]) / 2;
+    N[7] = r[0] * (r[0] - 1) * (1 - r[1] * r[1]) / 2;
+    N[8] = (1 - r[0] * r[0]) * (1 - r[1] * r[1]);
 }
 
 template <class T_X, class T_N>
-void ShapeQuad9::computeGradShapeFunction(const T_X &r, T_N &dNdr)
+void ShapeQuad9::computeGradShapeFunction(const T_X& r, T_N& dNdr)
 {
-    dNdr[8] = -2.0 * r[0] * (1.0 - r[1] * r[1]);
-    dNdr[7] = +0.5 * (1.0 - r[1] * r[1]) - 0.5 * dNdr[8];
-    dNdr[6] = -1.0 * r[0] * (1.0 - r[1]) - 0.5 * dNdr[8];
-    dNdr[5] = -dNdr[7];
```

Comment:
Finally, I found that dNdr[5] differs from ogs 5 and it is wrong. It is quit similar to  dNdr[7]
```
 -0.5 * (1.0 - r[1] * r[1]) - 0.5 * dNdr[8];
```
but it is not.